test/ibmq/test_ibmq_job.py

```diff
@@ -289,12 +292,87 @@ def test_retrieve_job_error(self, provider):
     def test_retrieve_jobs_status(self, provider):
         """Test retrieving jobs filtered by status."""
         backend = provider.get_backend('ibmq_qasm_simulator')
-        job_list = provider.backends.jobs(backend_name=backend.name(),
-                                          limit=5, skip=0, status=JobStatus.DONE)
+        # Get the most recent jobs that are done.
+        status_args = [JobStatus.DONE, 'DONE', [JobStatus.DONE], ['DONE']]
+        for arg in status_args:
+            with self.subTest(arg=arg):
+                backend_jobs = backend.jobs(limit=10, skip=0, status=arg)
+                self.assertTrue(backend_jobs)
+
+                for job in backend_jobs:
+                    self.assertTrue(job.status() is JobStatus.DONE,
+                                    "job status for job {} was '{}' but "
+                                    "it should be: '{}'"
+                                    .format(job.job_id(), job.status(), JobStatus.DONE))
 
-        self.assertTrue(job_list)
-        for job in job_list:
-            self.assertTrue(job.status() is JobStatus.DONE)
+    @requires_provider
+    def test_retrieve_all_job_statuses(self, provider):
```

Comment:
Honestly I don't see much value in this test case. All it shows is that _some_ jobs were returned from `.jobs()` (which is covered by other test cases). We can't really verify the content of the returned list.

Reply:
Yeah, I could see that. It seemed more relevant when we were comparing the two job lists retrieved, but then again you mentioned the chance of them being different if a job was submitted between the two `jobs` calls. I'll just remove it, since it would be very difficult to verify it as well.